salt/hosts-minion/init.sls

```diff
@@ -0,0 +1,14 @@
+{%- for minion_id, addrlist in salt['mine.get']('roles:kube-master', 'network.ip_addrs', 'grain').items() %}
+{{ minion_id }}-host-entry:
+  host.present:
+{% if addrlist is string %}
+    - ip: {{ addrlist }}
+{% else %}
+    - ip: {{ addrlist|first }}
+{% endif %}
+    - names:
+      - api
+      - api.{{ pillar['internal_infra_domain'] }}
+      - {{ minion_id }}
+      - {{ minion_id }}.{{ pillar['internal_infra_domain'] }}
+{% endfor %}
```

Comment:
Then keep also the `<hash>.<domain>` entries...

You'll have to use something like:

```
{% set api_servers = salt['mine.get']('roles:kube-master', 'network.ip_addrs', 'grain') %} 
{% set api_servers_addrs = api_servers.values() %}

api-host-entry:
  host.present:
{% if api_servers_addrs is string %}
    - ip: {{ api_servers_addrs }}
{% else %}
    - ip: {{ api_servers_addrs|first }}
{% endif %}
    - names:
      - api
      - api.{{ pillar['internal_infra_domain'] }}
{% endfor %}

{%- for server_id, addrlist in api_servers.items() %}
{{ server_id }}-host-entry:
   host.present:
{% if addrlist is string %}
     - ip: {{ addrlist }}
{% else %}
     - ip: {{ addrlist|first }}
{% endif %}
     - names:
       - {{ server_id }}
       - {{ server_id }}.{{ pillar['internal_infra_domain'] }}
{% endfor %}
```